### PR TITLE
Added size and date modified in listview

### PR DESCRIPTION
**Pull request by shiv07tiwari**
# Description

Added the size and date modified in pdf listview.
Screenshot added in issue.

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
# Checklist:
- [x ] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings

### CODE REVIEW

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 1]
```diff
@@ -148,8 +149,20 @@ public void onBindViewHolder(@NonNull ViewFilesHolder holder, int position) {
         final int filePosition = position;
         String[] name = fileName.split("/");
 
+        File file = new File(fileName);
+        Date lastModDate = new Date(file.lastModified());
+
+        String[] formatdate = lastModDate.toString().split(" ");
+        String time = formatdate[3];
+
+        String[] formattime =  time.split(":");
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Add a new function in fileutils to get the formatted date. And move all this code there

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 2]
```diff
@@ -148,8 +149,20 @@ public void onBindViewHolder(@NonNull ViewFilesHolder holder, int position) {
...
+        String date = formattime[0]+":"+formattime[1];
+
         holder.mFilename.setText(name[name.length - 1]);
 
+        holder.mFilesize.setText(String.format("%.2f MB", (double) file.length()/ (1024 * 1024)));
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: For size also, add a function in fileutils

**Comment by shiv07tiwari** [comment]
@Swati4star  Made the required changes.

**Comment by Swati4star** [comment]
https://travis-ci.org/Swati4star/Images-to-PDF/builds/397792007
@shiv07tiwari  Build & checkstyle, both are failing

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 3]
```diff
@@ -148,8 +150,14 @@ public void onBindViewHolder(@NonNull ViewFilesHolder holder, int position) {
         final int filePosition = position;
         String[] name = fileName.split("/");
 
+        File file = new File(fileName);
+        Date lastModDate = new Date(file.lastModified());
+
         holder.mFilename.setText(name[name.length - 1]);
 
+        holder.mFilesize.setText(FileUtils.getFormattedTime(file));
+        holder.mFiledate.setText(FileUtils.getFormattedDateandTime(lastModDate));
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: Pass `file` here & move 
```
File file = new File(fileName);
Date lastModDate = new Date(file.lastModified());
```

also to `getFormattedDateandTime` function

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 4]
```diff
@@ -103,4 +104,15 @@ public File getOrCreatePdfDirectory() {
         }
         return folder;
     }
+    public static String getFormattedDateandTime(Date lastdate){
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: Just like the other functions in this file, add comments to this one also.

**Comment by shiv07tiwari** [comment]
@Swati4star Build fail is due to checkstyle error, right ?
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:checkstyle'.
> Checkstyle rule violations were found. See the report at: file:///home/travis/build/Swati4star/Images-to-PDF/app/build/reports/checkstyle/checkstyle.html

**Comment by Swati4star** [comment]
No, it also failed in building:

```

* What went wrong:
Execution failed for task ':app:lintVitalRelease'.
> Lint found fatal errors while assembling a release target.
  
  To proceed, either fix the issues identified by lint, or modify your build script as follows:
  ...
  android {
      lintOptions {
          checkReleaseBuilds false
          // Or, if you prefer, you can continue to check for errors in release builds,
          // but continue the build even when errors are found:
          abortOnError false
      }
  }
  ...
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 2m 16s
41 actionable tasks: 41 executed
The command "./gradlew assembleDebug assembleRelease" exited with 1.
```

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -106,6 +106,8 @@
     <string name="example_pass_123">pass@123</string>
     <string name="remove_dialog">REMOVE</string>
     <string name="password_remove">Password Removed</string>
+    <string name="file_date_text">File Date</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: Add these strings to files of other locales also. That might fix the build error

Comment (shiv07tiwari) [reply app/src/main/res/values/strings.xml]: Okay.

**Comment by shiv07tiwari** [comment]
@Swati4star  What about the checkstyle error ?

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 6]
```diff
@@ -103,4 +104,24 @@ public File getOrCreatePdfDirectory() {
...
+        String time = formatdate[3];
+
+        String[] formattime =  time.split(":");
+        String date = formattime[0] + ":" + formattime[1];
+        return formatdate[0] + ", " + formatdate[1] + " " + formatdate[2] + " at " + date;
+    }
+    /**
+     * Gives a formatted time for pdf ListView
+     */
+    public static String getFormattedTime(File file) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: I don't think this function return formatted time. It returns formatted size.

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 7]
```diff
@@ -103,4 +104,24 @@ public File getOrCreatePdfDirectory() {
         }
         return folder;
     }
+    /**
+     * Gives a formatted date for pdf ListView
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: You haven't added the comments properly yet
```
/**
 * Gives a formatted last modified date for pdf ListView
 * @param file file object whose last modified date is to be returned
 *
 * @return String date modified in formatted form
 **/
```

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 8]
```diff
@@ -103,4 +104,24 @@ public File getOrCreatePdfDirectory() {
...
+     */
+    public static String getFormattedDate(File filename) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: change `filename` to `file`

File: app/src/main/java/swati4star/createpdf/util/FileUtils.java [hunk 9]
```diff
@@ -103,4 +104,24 @@ public File getOrCreatePdfDirectory() {
...
+
+        Date lastModDate = new Date(filename.lastModified());
+        String[] formatdate = lastModDate.toString().split(" ");
+        String time = formatdate[3];
+
+        String[] formattime =  time.split(":");
+        String date = formattime[0] + ":" + formattime[1];
+        return formatdate[0] + ", " + formatdate[1] + " " + formatdate[2] + " at " + date;
+    }
+    /**
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/FileUtils.java]: Write the comment properly.

**Comment by shiv07tiwari** [comment]
@Swati4star Added the changes.

File: app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java [hunk 10]
```diff
@@ -148,8 +150,14 @@ public void onBindViewHolder(@NonNull ViewFilesHolder holder, int position) {
         final int filePosition = position;
         String[] name = fileName.split("/");
 
+        File file = new File(fileName);
+        Date lastModDate = new Date(file.lastModified());
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java]: No need of these 2 lines now.

**Comment by shiv07tiwari** [comment]
@Swati4star  Squashed the commits.

**Comment by Swati4star** [comment]
Merged in a77c1aceb508cc7cf17fabde8d85b918d8b81bf9

**Pull request closed without merging**